src/lib/decorations/path.typ

```diff
@@ -59,6 +63,16 @@
   factor: 150%,
 )
 
+#let resolve-amplitude(amplitude, segment, num-segments) = {
+  return if type(amplitude) == function {
+    (amplitude)(segment / (num-segments - 1) * 100%)
+  } else if type(amplitude) == array {
+    amplitude.at(int(calc.max(0, calc.round(segment / (num-segments - 1) * (amplitude.len() - 1)))), default: 0)
```

Comment:
Same with this one, looking at test case:
```typc
wave(line((0,4), (4,4)), amplitude: (1,0,1))
```

Reply:
Changed it.